flower/utils/pg_storage.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+from datetime import datetime
+
+import pg8000
+
+logger = logging.getLogger(__name__)
+connection = None
+skip_callback = False
+
+_all_tables = """
+SELECT * FROM information_schema.tables
+WHERE table_schema = 'public'
+"""
+
+_schema = (
+    """CREATE TABLE events
+    (
+        id SERIAL PRIMARY KEY,
+        time TIMESTAMP,
+        data JSONB NOT NULL
+    )""",
+    "CREATE INDEX event_index ON events USING GIN (data)",
+    "CREATE INDEX event_time_index ON events (time ASC)",
+)
+
+_add_event = """INSERT INTO events (time, data) VALUES (%s, %s)"""
+
+_all_events = """SELECT data FROM events ORDER BY time ASC"""
+
+_ignored_events = {
+    'worker-offline',
+    'worker-online',
+    'worker-heartbeat',
+}
+
+
+def event_callback(state, event):
+    if skip_callback or event['type'] in _ignored_events:
+        return
+
+    cursor = connection.cursor()
+    try:
+        cursor.execute(_add_event, (
+            datetime.fromtimestamp(event['timestamp']),
+            json.dumps(event)
+        ))
+        connection.commit()
+    except:
+        connection.rollback()
+        raise
+    finally:
+        cursor.close()
+
+
+def open_connection(user, password, database, host, port, use_ssl):
+    global connection
+    connection = pg8000.connect(
+        user=user, password=password, database=database,
+        host=host, port=port, ssl=use_ssl
+    )
+
+    # Create schema if database is empty
+    cursor = connection.cursor()
+    try:
+        cursor.execute(_all_tables)
+        tables = cursor.fetchone()
```

Comment:
This doesn't work if any table already exists.  I ran into this because Azure's hosted postgresql has a table by default.  This should check for the table we care about instead.